### PR TITLE
Added QueueTask class.

### PR DESCRIPTION
**Pull request by drmonkeyninja**
To make it easier to develop custom tasks I've added a QueueTask class to extend that contains the basic properties and methods required for a queue task. The examples and readme have been updated accordingly.

I feel this makes it a little easier to start developing custom tasks as a beginner as it makes it simpler to see what is needed in order to write one.

### CODE REVIEW

File: Console/Command/Task/QueueTask.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Run function.
+ * This function is executed, when a worker is executing a task.
+ * The return parameter will determine, if the task will be marked completed, or be requeued.
+ *
+ * @param array $data The array passed to QueuedTask->createJob()
+ * @return bool Success
+ */
+	public function run($data) {
```
Comment (houseoftech) [inline Console/Command/Task/QueueTask.php]: Good idea to add the class.

Can the $id attribute be added back? It is passed to the run function by QueueShell.

**Comment by drmonkeyninja** [comment]
Thanks houseoftech. My bad with the $id attribute, I appear to have removed it by mistake. Will restore.

**Comment by dereuromark** [comment]
You have a few more lines of wrong changes in your diff. It would be nice if those were also reverted.

**Comment by drmonkeyninja** [comment]
Hi Mark, which lines in particular are you referring to? It all looks fine to me now. I've removed a few of the example task properties as these are now getting inherited. Perhaps I'm missing something when I look at the comparison.

File: Console/Command/Task/QueueExampleTask.php [hunk 2]
```diff
@@ -89,10 +75,9 @@ public function add() {
  * The return parameter will determine, if the task will be marked completed, or be requeued.
  *
  * @param array $data The array passed to QueuedTask->createJob()
- * @param int $id The id of the QueuedTask
  * @return bool Success
  */
-	public function run($data, $id = null) {
```
Comment (dereuromark) [inline Console/Command/Task/QueueExampleTask.php]: here

File: Console/Command/Task/QueueExecuteTask.php [hunk 3]
```diff
@@ -89,7 +75,6 @@ public function add() {
  * The return parameter will determine, if the task will be marked completed, or be requeued.
  *
  * @param array $data The array passed to QueuedTask->createJob()
- * @param int $id The id of the QueuedTask
```
Comment (dereuromark) [inline Console/Command/Task/QueueExecuteTask.php]: and here (see failing CS report)

File: Console/Command/Task/QueueTask.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	public $autoUnserialize = true;
+
+/**
+ * Add functionality.
+ *
+ * @return void
+ */
+	public function add() {
+
+		return;
```
Comment (dereuromark) [inline Console/Command/Task/QueueTask.php]: there should be no whitespace newlines at the beginning or end of a method.

**Comment by drmonkeyninja** [comment]
Sorry about that. Should be fixed now.

**Comment by dereuromark** [comment]
Please (always) double check the diff.

File: Console/Command/Task/QueueExecuteTask.php (hunk 3 above)
Comment (dereuromark) [inline Console/Command/Task/QueueExecuteTask.php]: !

**Comment by drmonkeyninja** [comment]
Normally am good at checking the diffs, but must have been having a bit of an off day today. Hope everything is fine with the pull request now.

**Comment by dereuromark** [comment]
See my `!` above :)

**Comment by dereuromark** [comment]
Thx its a nice cleanup!

**Pull request merged**